src/app/_components/BlockList/LayoutA/useBlockList.ts

```diff
@@ -61,144 +62,93 @@ const createUIBlockList = (
-          setLatestBlocks(prevLatestBlocks => {
-            const updatedList = { ...prevLatestBlocks };
-            updatedList[block.hash] = block;
-            return updatedList;
-          });
-        }
-
-        if (liveUpdates) {
-          setFadeEffect(true);
-          setTimeout(() => {
-            updateLatestBlocks();
-            setFadeEffect(false);
-          }, 500);
-        } else {
-          updateLatestBlocks();
-        }
-      });
-    };
-
-    void subscribe();
-
-    return () => {
-      if (subRef.current?.unsubscribe) {
-        void subRef.current?.unsubscribe();
-      }
-    };
-  }, [liveUpdates]);
-
-  const clearLatestBlocks = () => {
-    setLatestBlock(undefined);
-    setLatestBlocks({});
-  };
-
-  return { latestBlocks, clearLatestBlocks, latestBlock };
-}
-
-export function useBlockListData() {
-  const burnBlocks = useSuspenseInfiniteQueryResult<BurnBlock>(
-    useSuspenseBurnBlocks(BURN_BLOCK_LENGTH),
-    BURN_BLOCK_LENGTH
-  );
-
-  const lastBurnBlock = burnBlocks[0];
-  const secondToLastBurnBlock = burnBlocks[1];
-
-  const lastBurnBlockStxBlocks = useSuspenseInfiniteQueryResult(
-    useSuspenseBlocksByBurnBlock(lastBurnBlock.burn_block_height, STX_BLOCK_LENGTH),
-    STX_BLOCK_LENGTH
-  );
-  const secondToLastBlockStxBlocks = useSuspenseInfiniteQueryResult(
-    useSuspenseBlocksByBurnBlock(secondToLastBurnBlock.burn_block_height, STX_BLOCK_LENGTH),
-    STX_BLOCK_LENGTH
-  );
+export function useBlockList(length: number) {
+  const queryClient = useQueryClient();
+  const { setFadeEffect, liveUpdates } = useBlockListContext();
 
-  return {
+  const {
     lastBurnBlock,
     secondToLastBurnBlock,
     lastBurnBlockStxBlocks,
     secondToLastBlockStxBlocks,
-  };
-}
+  } = useInitialBlockList();
+
+  const initialBlockHashes = useMemo(
+    () =>
+      new Set([
+        ...lastBurnBlockStxBlocks.map(block => block.hash),
+        ...secondToLastBlockStxBlocks.map(block => block.hash),
+      ]),
+    [lastBurnBlockStxBlocks, secondToLastBlockStxBlocks]
+  );
 
-export function useBlockList(
-  length: number,
-  liveUpdates: boolean,
-  setFadeEffect: (value: boolean) => void
-) {
-  const queryClient = useQueryClient();
-  const [displayedBlocks, setDisplayedBlocks] = React.useState<Record<string, number>>({});
-  const { latestBlocks, latestBlock, clearLatestBlocks } = useBlockListWebSocket(
-    liveUpdates,
-    setFadeEffect
+  const initialBurnBlockHashes = useMemo(
+    () => new Set([lastBurnBlock.burn_block_hash, secondToLastBurnBlock.burn_block_hash]),
+    [lastBurnBlock, secondToLastBurnBlock]
   );
 
-  const refetch = useCallback(
-    function () {
-      clearLatestBlocks();
-      return Promise.all([
+  const { latestBlock, latestBlocksCount, clearLatestBlocks } = useBlockListWebSocket(
+    initialBlockHashes,
+    initialBurnBlockHashes
+  );
+
+  const updateList = useCallback(
+    async function () {
+      setFadeEffect(true);
```

Comment:
Couldnt this just be a setIsUpdateListLoading? and then use that to trigger the fade effect?

Reply:
Could you elaborate more on that?

Reply:
The name setFadeEffect seems too specific. What you are actually tracking is if the update list operation is loading. Why not then change setFadeEffect to setIsUpdateListLoading as that would communicate a lot more information about why there is a fade effect. Additionally, in the future, if I needed to use this state to trigger a different operation, triggering it off of teh state of the fade effect wouldn't make any semantic sense